src/types/mod.rs

```diff
@@ -277,6 +277,10 @@ pub enum UnsolicitedResponse {
     /// sequence numbers 9, 8, 7, 6, and 5.
     // TODO: the spec doesn't seem to say anything about when these may be received as unsolicited?
     Expunge(Seq),
+    /// An uncategorised response - one which has no code associated with it, just a text
```

Comment:
Let's try to be a little more helpful here to potentially-confused receivers of this code. How about:
```rust
/// An unsolicited [`OK` response](https://tools.ietf.org/html/rfc3501#section-7.1.1) that
/// indicates an information-only message; the nature of the information MAY be
/// indicated by a response code.
///
/// The untagged OK is also used as one of three possible greetings
/// at connection startup.  It indicates that the connection is not
/// yet authenticated and that a `LOGIN` command is needed.
Notice(String)
```